src/components/movies/movies.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+
+import Modal from '../modal/modal';
+import ModalAddEditContent, { AddEditFormData } from '../modal/modal-add-edit-content';
+import ModalDeleteContent from '../modal/modal-delete-content';
+import Filter from './filter/filter';
+import MovieCard from './movie-card/movie-card';
+import { MoviesContainer } from './movies-container';
+import { MoviesResults } from './movies-results';
+import { MoviesResultsNumber } from './movies-results-number';
+
+export interface Movie {
+  title: string;
+  image: string;
+  genre: string;
+  date: number;
+  id: string;
+}
+
+const movies: Movie[] = [
+  {
+    title: "How to train your dragon",
+    image: "src/assets/images/how-to-train-your-dragon.jpg",
+    genre: "cartoon",
+    date: 2019,
+    id: "rick001",
+  },
+  {
+    title: "Harry Potter",
+    image: "src/assets/images/HP.jpg",
+    genre: "cartoon",
+    date: 2019,
+    id: "rick002",
+  },
+  {
+    title: "Khumba",
+    image: "src/assets/images/khumba.jpg",
+    genre: "cartoon",
+    date: 2019,
+    id: "rick003",
+  },
+  {
+    title: "Movie",
+    image: "src/assets/images/movie-1.jpg",
+    genre: "cartoon",
+    date: 2019,
+    id: "rick004",
+  },
+  {
+    title: "How to train your dragon",
+    image: "src/assets/images/how-to-train-your-dragon.jpg",
+    genre: "cartoon",
+    date: 2019,
+    id: "rick005",
+  },
+  {
+    title: "Harry Potter",
+    image: "src/assets/images/HP.jpg",
+    genre: "cartoon",
+    date: 2019,
+    id: "rick006",
+  },
+  {
+    title: "Khumba",
+    image: "src/assets/images/khumba.jpg",
+    genre: "cartoon",
+    date: 2019,
+    id: "rick007",
+  },
+  {
+    title: "Movie",
+    image: "src/assets/images/movie-1.jpg",
+    genre: "cartoon",
+    date: 2019,
+    id: "rick008",
+  },
+];
+
+const form: AddEditFormData = {
+  gendre: "horror",
+  title: "Title 1",
+  overview: "Overview 1",
+  movieUrl: "movie url 1",
+  runTime: "runtime 1",
+  releaseDate: "02-09-2020",
+};
+
+const filter = ["all", "documentary", "comedy", "horror", "crime"];
+const sort = ["Date", "Genre", "Title"];
+const menuOptions = ["Edit", "Delete"];
+
+const MoviesComponent = (props: { className: string }) => {
```

Comment:
This component should called `MoviesContainer` since it is fetching data and passing it to the child components.
Read about container and presentational components. Despite this terms became a little bit deprecated, it will be useful to know about this type of components splitting in React.
[https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0](url)